### PR TITLE
[SPARK-41721][CONNECT][TESTS] Enable doctests in pyspark.sql.connect.catalog

### PR DESCRIPTION
**Pull request by HyukjinKwon**
### What changes were proposed in this pull request?

This PR proposes to enable doctests in `pyspark.sql.connect.catalog` that is virtually the same as `pyspark.sql.catalog`.

### Why are the changes needed?

To make sure on the PySpark compatibility and test coverage.

### Does this PR introduce _any_ user-facing change?

No, test-only.

### How was this patch tested?

CI in this PR should test this out.

### CODE REVIEW

File: python/pyspark/sql/catalog.py [hunk 1]
```diff
@@ -273,15 +273,15 @@ def databaseExists(self, dbName: str) -> bool:
 
         >>> spark.catalog.databaseExists("test_new_database")
         False
-        >>> _ = spark.sql("CREATE DATABASE test_new_database")
+        >>> _ = spark.sql("CREATE DATABASE test_new_database").collect()
```
Comment (HyukjinKwon) [inline python/pyspark/sql/catalog.py:276]: Hmmm ... this is orthogonal issue but just to make sure we don't forget. I think we should make `sql` method in Spark Connect to be analyzed per every call. Otherwise, we can't make it compatible with the existing PySpark usage.

I changes the existing doctest here for now to make the tests pass but should change this back ideally.

Comment (HyukjinKwon) [reply python/pyspark/sql/catalog.py:276]: cc @hvanhovell @grundprinzip FYI

**Review by zhengruifeng — APPROVED** [review]

File: python/pyspark/sql/catalog.py (hunk 1 above)
Comment (zhengruifeng) [reply python/pyspark/sql/catalog.py:276]: what about create a jira ticket for this (should change this back ideally)?

Comment (HyukjinKwon) [reply python/pyspark/sql/catalog.py:276]: 👌

**Comment by HyukjinKwon** [comment]
Merged to master.
